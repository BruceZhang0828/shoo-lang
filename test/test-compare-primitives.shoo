int a = 5;
int b = 5;

if (a == b) {
    println("int works");
} else {
    println("int doesn't work");
}

int c = 6;

if (a == c) {
    println("int doesn't work");
} else {
    println("int works");
}

if (a == a) {
    println("int works");
} else {
    println("int doesn't work");
}

// floats
float d = 1.098;
float e = 1.098;

if (d == e) {
    println("float works");
} else {
    println("float doesn't work");
}

float f = 2.45678900;
if (d == f) {
    println("float doesn't work");
} else {
    println("float works");
}

float g = 1.0980;
if (d == g) {
    println("float works");
} else {
    println("float doesn't work");
}

if (d == d) {
    println("float works");
} else {
    println("float doesn't work");
}

// booleans
bool h = true;
bool i = true;

if (h == i) {
    println("bool works");
} else {
    println("bool doesn't work");
}

bool j = false;
if (h == j) {
    println("bool doesn't work");
} else {
    println("bool works");
}

bool k = 0 < 5;
if (h == k) {
    println("bool works");
} else {
    println("bool doesn't work");
}

if (h == h) {
    println("bool works");
} else {
    println("bool doesn't work");
}

