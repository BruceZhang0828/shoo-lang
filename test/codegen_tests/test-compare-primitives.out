int works
int works
int works
float works
float works
float works
float works
bool works
bool works
bool works
bool works
